src/reports/io/hdf5_writer.cpp

```diff
@@ -0,0 +1,140 @@
+#include <iostream>
+#include <tuple>
+
+#include "hdf5_writer.hpp"
+#include <reports/library/implementation_interface.hpp>
+#include <reports/library/sonatareport.hpp>
+
+HDF5Writer::HDF5Writer(const std::string& report_name)
+    : IoWriter(report_name)
+    , m_file(0)
+    , m_dataset(0)
+    , m_collective_list(0)
+    , m_independent_list(0) {
+    hid_t plist_id = H5Pcreate(H5P_FILE_ACCESS);
```

Comment:
We choose pure HDF5 because initially we needed support for compound datatypes (not supported in HighFive yet).
We don't use them yet, but it is expected than in the future we will.

Reply:
Since we don't need them now, and since all (?) the functionality exists in `HighFive`, it would be best if we continue to use just that.